src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1603,112 +1646,98 @@ public RefreshItem(String keyInCache, AuthenticationRequest request, TokenCacheI
+            Logger.v(TAG, "User is not provided, no need to look for FRT.");
             return null;
         }
         
-        if (userIdentifierType == UserIdentifierType.UniqueId) {
-            Logger.v(TAG + methodName, "UserIdentifier type is unique id, will look for the family item based on unique id.");
-            while (allItems.hasNext()) {
-                final TokenCacheItem tokenCacheItem = allItems.next();
-                final UserInfo tokenUserInfo = tokenCacheItem.getUserInfo();
-                if (tokenUserInfo != null && userId.equalsIgnoreCase(tokenUserInfo.getUserId())) {
-                    if (!StringExtensions.IsNullOrBlank(tokenCacheItem.getFamilyClientId())
-                            && !StringExtensions.IsNullOrBlank(tokenCacheItem.getRefreshToken())) {
-                        Logger.v(TAG + methodName, "Found family item matching the given user id, and the clientId selected for FoCI is: " + tokenCacheItem.getClientId());
-                        return tokenCacheItem;
-                    }
-                }
-            }
-        } else if (userIdentifierType == UserIdentifierType.LoginHint) {
-            Logger.v(TAG + methodName, "UserIdentifier type is loginhint, will look for the family item based on login hint.");
-            while (allItems.hasNext()) {
-                final TokenCacheItem tokenCacheItem = allItems.next();
-                final UserInfo tokenUserInfo = tokenCacheItem.getUserInfo();
-                if (tokenUserInfo != null && userId.equalsIgnoreCase(tokenUserInfo.getDisplayableId())) {
-                    if (!StringExtensions.IsNullOrBlank(tokenCacheItem.getFamilyClientId())
-                            && !StringExtensions.IsNullOrBlank(tokenCacheItem.getRefreshToken())) {
-                        Logger.v(TAG + methodName, "Found family item matching the given user id, and the clientId selected for FoCI is: " + tokenCacheItem.getClientId());
-                        return tokenCacheItem;
-                    }
-                }
-            }
+        final String keyUsed = CacheKey.createFamilyRefreshTokenKey(authRequest, userId);
+        final TokenCacheItem familyTokenCacheItem = mTokenCacheStore.getItem(keyUsed);
+        
+        return createRefreshItem(keyUsed, authRequest, familyTokenCacheItem);
+    }
+    
+    /**
+     * @return Either unique user id or displayable id if provided in the request.
+     */
+    private String getUserFromAuthenticationRequest(final AuthenticationRequest authRequest) {
+        String userId = authRequest.getUserId();
+        if (StringExtensions.IsNullOrBlank(userId)) {
+            userId = authRequest.getLoginHint();
         }
+        
+        return userId;
+    }
+    
+    /**
+     * @return {@link RefreshItem}.
+     */
+    private RefreshItem createRefreshItem(final String keyUsed, final AuthenticationRequest authRequest, 
```

Comment:
NIT: find, create, get - let's be consistent with our method names... I expect a create* method to fail only in some corner cases such as out of memory. Get OTOH will return null if the item/thing does not exist in the store.